src/pfe/portal/modules/User.js

```diff
@@ -659,6 +670,288 @@ module.exports = class User {
+   */
+  async getRegistrySecretList() {
+    const registrySecretList = [];
+    try {
+      const isDockerConfigFilePresent = await cwUtils.fileExists(this.dockerConfigFile)
+      if (isDockerConfigFilePresent) {
+        log.info("Docker Config file present, returning the Docker Registry List");
+        const jsonObj = await fs.readJson(this.dockerConfigFile);
+
+        for (let key in jsonObj.auths) {
+          const registrySecret = {
+            url: key,
+            username: jsonObj.auths[key].username
+          };
+          registrySecretList.push(registrySecret);
+        }
+      } else {
+        log.info("No Docker Config file present, no Docker Registry List to return");
+      }
+    
+      log.debug("Codewind Docker Registry List: " + JSON.stringify(registrySecretList));
+    } catch (err) {
+      const msg = "Failed to get the Codewind Docker Config Registries";
+      log.error(err);
+      log.error(msg);
+      const retval = {
+        statusCode: 500,
+        body: msg
+      };
+      return retval;
+    }
+    
+    const retval = {
+      statusCode: 200,
+      body: registrySecretList
+    };
+    return retval;
+  }
+
+  /**
+   * Function to remove the docker registries from PFE
+   */
+  async removeRegistrySecret(url) {
+    let createKubeSecret = false;
+    const registrySecretList = [];
+
+    try {
+      const isDockerConfigFilePresent = await cwUtils.fileExists(this.dockerConfigFile)
+      if (isDockerConfigFilePresent) {
+        log.info("Docker Config file present, removing the specified Docker Registry from the list");
+        const jsonObj = await fs.readJson(this.dockerConfigFile);
+
+        for (let key in jsonObj.auths) {
+          if (key == url) {
+            delete jsonObj.auths[key];
+          } else {
+            const registrySecret = {
+              url: key,
+              username: jsonObj.auths[key].username
+            };
+            registrySecretList.push(registrySecret);
+          }
+        }
+
+        await fs.writeJson(this.dockerConfigFile, jsonObj);
+        createKubeSecret = true;
+        log.info("The Docker config file has been updated for removal of " + url);
+      } else {
+        // return if there is no Docker Config file, no need to create new Kubernetes Secret or patch the Service Account
+        const msg = "No Docker Config file present, no Docker Registry to remove from the list";
+        log.error(msg);
+        const retval = {
+          statusCode: 400,
+          body: msg
+        };
+        return retval;
+      }
+    } catch (err) {
+      const msg = "Failed to update the Codewind Docker Config File";
+      log.error(err);
+      log.error(msg);
+      const retval = {
+        statusCode: 500,
+        body: msg
+      };
+      return retval;
+    }
+
+    // Only update the Kube secret when running in K8s
+    let isServiceAccountPatched = false;
+    if (createKubeSecret && global.codewind.RUNNING_IN_K8S) {
```

Comment:
As discussed should check what we want to do for the non-K8s scenario and test it out to make sure the response is what we'd expect.

Reply:
updated